.github/workflows/backend-formatting.yml

```
@@ -0,0 +1,55 @@
+name: rust-formatting
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  run-rustfmt:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install rust-toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+          components: clippy, rustfmt
+      - name: Run rustfmt using cargo
+        working-directory: dtm-database-backend
+        run: cargo fmt
+      - name: Commit formatted code
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: backend's code formatted by rust-formatting CI
+
+  run-cargo-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install rust-toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+          components: clippy, rustfmt
+
+      - name: Run cargo-test using cargo
+        working-directory: dtm-database-backend
+        run: cargo test
+
+  run-cargo-clippy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install rust-toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+          components: clippy, rustfmt
+
+      - name: Run cargo-test using cargo
+        working-directory: dtm-database-backend
+        run: cargo clippy
```

```suggestion
      - name: Run cargo-test using cargo
        working-directory: dtm-database-backend
        run: cargo clippy -- -Dwarnings
```
How about making results fail even if only warnings have appeared?

---

-Dwarning? I don't know that. However that is good idea!